stream-chat-android-client/src/test/java/io/getstream/chat/android/client/clientstate/ClientStateServiceTests.kt

```diff
@@ -25,6 +25,25 @@ internal class ClientStateServiceTests {
         sut.state shouldBeEqualTo ClientState.Idle
     }
 
+    @Test
+    fun `Given Idle state When disconnected requested Should stay in Idle state`() {
+        val sut = Fixture().please()
+
+        sut.onDisconnectRequested()
+
+        sut.state shouldBeEqualTo ClientState.Idle
+    }
+
+    @Test
+    fun `Given Idle state When connected Should stay in Idle state`() {
```

Comment:
Is the "Connected" event in this state invalid (we shouldn't have received it), so we're ignoring it?

Reply:
Discussed this offline, we're not 100% sure why this event happens in this case. This will fix the crash that it produces, and we can keep investigating this later.